### DISCUSSION TITLE
5.1.4 ACL no longer working

### DISCUSSION THREAD

**Discussion by ares003** [phalcon/phalcon#329] [Q&A]
Hello,
I inherited a project that utilizes phalcon and I had to upgrade from 3.x to 5.1.4 due to upgrading PHP for security compliance. I have managed to get most of the project functioning again, however I have issues with the ACL not working. I don't understand how it is supposed to work to begin (even after reading the documentation) with so I'm struggling to discover why it stopped working. Has anybody else run into this situation with the upgrade? no matter what it just says that I don't have access.
Thanks in advance for any advice.

**Comment by nask0** [comment]
I didn't experience the same problem, but however, when I find myself in similar situation, as final resort I  take a look at the code, as I often found, it's the place when things are "most up-to-date". You can check out https://github.com/phalcon/cphalcon/tree/5.0.x/phalcon/Acl - code is pretty similar to PHP so it's readable enough.
The other thing is to take a look at your implementation again, maybe there is some breaking change or else.

**Comment by kgrammer** [comment]
Head over to the Phalcon channel on Discord (https://phalcon.io/discord).  It has more active discussions so you might find more people there who can assist with your issue.
